lambdas/tests/unit/services/security/jwe-decrypter.test.ts

```diff
@@ -0,0 +1,82 @@
+import { createDecipheriv } from "crypto";
+import { KMSClient } from "@aws-sdk/client-kms";
+import { JweDecrypter } from "../../../../src/services/security/jwe-decrypter";
+
+jest.mock("crypto", () => ({
+  createDecipheriv: jest.fn().mockReturnValueOnce({
+    setAuthTag: jest.fn(),
+    setAAD: jest.fn(),
+    update: jest.fn().mockReturnValueOnce(Buffer.from("decrypted content")),
+    final: jest.fn(),
+  }),
+}));
+
+jest.mock("@aws-sdk/client-kms", () => ({
+  KMSClient: jest.fn(() => ({
+    send: jest.fn().mockResolvedValue({
+      Plaintext: "decryptedContentEncKey",
+    }),
+  })),
+  DecryptCommand: jest.fn(() => ({})),
+  EncryptionAlgorithmSpec: {
+    RSAES_OAEP_SHA_256: 'RSAES_OAEP_SHA_256',
+  },
+}));
+
+jest.mock("jose", () => ({
+  base64url: {
+    decode: jest.fn().mockReturnValue(new Uint8Array([1, 2, 3, 4])),
+  },
+}));
+
+const decryptedContentEncKey = "decryptedContentEncKey";
+describe("JweDecrypter", () => {
+  let kmsClient: KMSClient;
+  let jweDecrypter: JweDecrypter;
+  const getEncryptionKeyId = jest.fn();
+  const jweProtectedHeader = {
+    alg: 'RSA-OAEP',
+    enc: 'A256GCM',
+    kid: 'kid',
+  };
+  const compactJwe = `${jweProtectedHeader}.${"encryptedKey"}.${"iv"}.${"ciphertext"}.${"tag"}`;
+
+  beforeEach(() => {
+    kmsClient = new KMSClient({});
+    jweDecrypter = new JweDecrypter(kmsClient, getEncryptionKeyId.mockReturnValueOnce('test-key-id'));
+    jest.spyOn(JSON, 'parse').mockReturnValueOnce(jweProtectedHeader);
+  });
+
+  it("decrypts JWE", async () => {
+    const decoded_iv = new Uint8Array([1, 2, 3, 4]);
+    const response = {
+      Plaintext: "decryptedContentEncKey"
+    };
+    const kmsClientMock = <jest.Mock>kmsClient.send;
+    kmsClientMock.mockResolvedValue(response);
+    const result = await jweDecrypter.decryptJwe(compactJwe);
+
+    expect(getEncryptionKeyId).toHaveBeenCalled();
```

Comment:
These are three different checks, so I'd break these out into separate checks. 

If necessary I'd pull out the common code into a `beforeEach` and if it wasn't relevant to the other tests, I'd nest the `beforeEach` inside a `describe` or `context` block.


```
describe('JweDecrypter', ()=>{
  describe('#decriptJwe', ()=>{
    it('should call getEncryptionId',()=>{});
    it('should createDecipheriv with algorithm and key properties',()=>{});
    it('should decrypt jwe value',()=>{});
  });
});
```

It seems like a lot of structure, but it does allow you to iterate in a TDD way really quickly.